src/core/storage.rs

```diff
@@ -133,10 +133,34 @@ pub struct RecordsStorageConfig {
 	pub max_blocks: usize,
 }
 
+/// This trait defines basic functions for the storage
+pub trait RecordsStorage<K> {
+	/// Creates a new storage with the specified config
+	fn new(config: RecordsStorageConfig) -> Self;
+	/// Inserts a record in ephemeral storage. This method does not overwrite
+	/// records and returns an error in case of a duplicate entry.
+	fn insert(&mut self, key: K, entry: StorageEntry) -> color_eyre::Result<()>;
```

Comment:
what happens if the key already exists ?

Reply:
An error is returned, as we discussed, the keys must be distinct, so a duplicate is likely an error in the logic of the code.